stablehlo/dialect/TypeInference.cpp

```diff
@@ -2708,12 +2708,13 @@ LogicalResult inferTransposeOp(Optional<Location> loc, Value operand,
 
 LogicalResult inferTriangularSolveOp(
     Optional<Location> location, Value a, Value b, bool leftSide,
-    bool isTransposeAInvalid,
+    bool isTransposeAInvalid, bool isTransposeOrAdjoint,
     SmallVectorImpl<ShapedTypeComponents>& inferredReturnShapes) {
   // ODS enforces that a and b are of same element type: float or complex.
   auto elementType = a.getType().cast<ShapedType>().getElementType();
   auto aType = a.getType().dyn_cast<RankedTensorType>();
-  if (!aType) {
+  auto bType = b.getType().dyn_cast<RankedTensorType>();
+  if (!aType || !bType) {
```

Comment:
I don't think we should move the `bType` check here because then we'll regress verification for the case when `aType` is ranked but `bType` is unranked.